ResearchKit/Common/ORKSurveyAnswerCellForText.m

```diff
@@ -327,12 +351,15 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
 }
 
 - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
-    return YES;
+    BOOL isValid = [self isAnswerValidForTextField:textField];
+    return isValid;
 }
 
 - (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    BOOL isValid = [self isAnswerValidForTextField:textField];
+    
     [self.textField resignFirstResponder];
-    return YES;
+    return isValid;
```

Comment:
Replace with `return [self isAnswerValidForTextField:textField]` and remove line 359.